### ISSUE TITLE
Implement perceptron with convergence to increase accuracy and training efficiency

### ISSUE THREAD

**Issue by adster283** [adster283/neural-networks#1]
In the current state the perceptron is both over training and over correcting weights after every epoch. This model needs to be changed to implement a better convergence and to better track the learning rate as to not over train on the training data set.
